packages/wonder-blocks-modal/components/two-column-modal.js

```diff
@@ -35,33 +42,73 @@ type Props = {
 export default class TwoColumnModal extends React.Component<Props> {
     static defaultProps = {
         onClickCloseButton: () => {},
+        smallSidebar: false,
     };
 
     render() {
+        const {sidebar, footer, contents, smallSidebar} = this.props;
+
+        const sidebarView = smallSidebar ? (
+            <Cell
+                smCols={4}
+                largeCols={1}
+                style={[styles.column, styles.bgSidebar, styles.bgSidebarSmall]}
+            >
+                {sidebar}
+            </Cell>
+        ) : (
+            <FlexCell style={[styles.column, styles.bgSidebar]}>
+                <View style={styles.sidebar}>{sidebar}</View>
+            </FlexCell>
+        );
+
         return (
-            <View style={styles.container}>
+            <View
+                style={[
+                    styles.container,
+                    !sidebar && styles.containerNoSidebar,
+                ]}
+            >
+                <Grid spec={GRID_MODAL_SPEC}>
+                    <Row flush>
+                        {sidebar && sidebarView}
+                        <FlexCell
+                            style={[
+                                styles.column,
+                                styles.bgContents,
+                                !sidebar && styles.bgContentsNoSidebar,
+                            ]}
+                        >
+                            <View
+                                style={[
+                                    styles.contents,
+                                    smallSidebar && styles.contentsSmall,
+                                    !!footer && styles.contentsHasFooter,
+                                    !sidebar && styles.contentsNoSidebar,
+                                ]}
+                            >
+                                {contents}
+                            </View>
+                            {footer && (
+                                <View style={styles.footer}>{footer}</View>
+                            )}
+                        </FlexCell>
+                    </Row>
+                </Grid>
                 <View style={styles.closeButton}>
                     <ModalCloseButton
-                        color="light"
+                        color={sidebar ? "light" : "dark"}
                         onClick={this.props.onClickCloseButton}
                     />
                 </View>
-                <View style={[styles.column, styles.leftColumn]}>
-                    {this.props.leftContent}
-                </View>
-                <View style={[styles.column, styles.rightColumn]}>
-                    {this.props.rightContent}
-                </View>
             </View>
         );
     }
 }
 
 const styles = StyleSheet.create({
     container: {
-        display: "flex",
-        flexDirection: "row",
-        alignItems: "stretch",
+        margin: "0 auto",
```

Comment:
Curious what this self-centering rule is for.

Reply:
I did this to center the modal inside the container. Although maybe I shouldn't be doing that and should leave it up to the container to take care of? Thoughts?

Reply:
Yeah, shrug! I would generally expect the modal to handle alignment itself, but this isn't necessarily a ridiculous default. (Then again, I'm not sure how easy it is for parents to override things like `margin: 0 auto`—if it's tricky, maybe it's worth erring toward giving the parent more freedom.)